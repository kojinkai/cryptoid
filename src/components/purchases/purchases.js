// @flow
import React, { Component } from 'react';
import { format } from 'date-fns';
import { getPrettyCurrencyFromCode } from '../../helpers';
import './purchases.css';

class Purchases extends Component {
  render() {
    const { purchases, activeWalletName } = this.props;
    const transactionDescription = `Bought ${getPrettyCurrencyFromCode(activeWalletName)}`;

    const formattedPurchases = purchases.map(purchase => {
      return (
        <div key={purchase.id} className="purchases__purchase-wrapper">
          <div className="purchases__purchase-date">
            <span>{format(new Date(purchase.created_at), 'MMM')}</span>
            <span>{format(new Date(purchase.created_at), 'D')}</span>
          </div>
          <div className="purchases__amount-bought">
            {purchase.amount.amount}
          </div>
          <div className="purchases__transaction-info">
            <span className="purchases__transaction-type">
              {transactionDescription}
            </span>
          </div>
          <div className="purchases__amount-paid">
            {purchase.total.amount}
          </div>
        </div>
      )
    }
    );

    return (
      <section className="purchases">
        {formattedPurchases}
      </section>
    );
  }
}


export default Purchases;