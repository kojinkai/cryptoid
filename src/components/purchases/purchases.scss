@import 'styles/colors';
@import 'styles/grid';
@import 'styles/breakpoints';

.purchases {

}
