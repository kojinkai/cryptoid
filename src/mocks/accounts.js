export default [
  {
    'balance': {
      'amount': '800',
      'currency': 'GBP'
    },
    'currency': 'BTC',
    'id': '58542935-67b5-56e1-a3f9-42686e07fa40',
    'name': 'BTC'
  },
  {
    'balance': {
      'amount': '395.90',
      'currency': 'USD'
    },
    'currency': 'BTC',
    'id': '2bbf394c-193b-5b2a-9155-3b4732659ede',
    'name': 'ETH'
  },
  {
    'balance': {
      'amount': '400.00',
      'currency': 'USD'
    },
    'currency': 'GBP',
    'id': '1ggf398d-193b-5b2a-9155-3b4732659ede',
    'name': 'GBP'
  }  
];
