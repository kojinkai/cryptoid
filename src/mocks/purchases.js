export default [
  {
    'currency': 'BTC',
    'id': '9e14d574-30fa-5d85-b02c-6be0d851d61d',
    'total': {
      'amount': '101.01',
      'currency': 'GBP'
    }
  },
  {
    'currency': 'BTC',
    'id': '8e14d574-30fa-5d85-b02c-6be0d851d61d',
    'total': {
      'amount': '202.02',
      'currency': 'GBP'
    }
  },
  {
    'currency': 'BTC',
    'id': '8e14d574-30fa-5d85-b02c-6be0d851d61d',
    'total': {
      'amount': '303.03',
      'currency': 'GBP'
    }
  }
];