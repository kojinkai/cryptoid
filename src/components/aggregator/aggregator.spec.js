import React from 'react';
import { shallow } from 'enzyme';
import Aggregator from './aggregator';
import purchases from '../../mocks/purchases';
import accounts from '../../mocks/accounts';

describe('when rendering the aggregator component', () => {
  it('it should render an aggregate price', () => {
    const wrapper = shallow(<Aggregator account={accounts[0]} purchases={purchases} isLoading={ false } />);
    expect(wrapper.find('.aggregator__pl-currency').find('span').text()).toBe('£193.94');
  });
  it('it should render an total paid price', () => {
    const wrapper = shallow(<Aggregator account={accounts[0]} purchases={purchases} isLoading={ false } />);
    expect(wrapper.find('.aggregator__metadata').childAt(0).text()).toBe('£606.06 paid');
  });
  it('it should render a balance now', () => {
    const wrapper = shallow(<Aggregator account={accounts[0]} purchases={purchases} isLoading={ false } />);
    expect(wrapper.find('.aggregator__metadata').childAt(1).text()).toBe('£800 now');
  });
  it('it should render a percentage tilt', () => {
    const wrapper = shallow(<Aggregator account={accounts[0]} purchases={purchases} isLoading={ false } />);
    expect(wrapper.find('.aggregator__metadata').childAt(2).text()).toBe('+132.00%');
  });
});
