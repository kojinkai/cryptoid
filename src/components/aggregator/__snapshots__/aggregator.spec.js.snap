// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Aggregator /> should render according the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="aggregator"
>
    <div
        className="aggregator__container"
    >
        <h1
            className="aggregator__pl-currency"
        >
            <Indicator
                className="aggregator__pl-indicator"
                inProfit={true}
            />
            <span>
                £193.94
            </span>
        </h1>
        <div
            className="aggregator__metadata"
        >
            <span>
                £606.06 paid
            </span>
            <span>
                £800 now
            </span>
            <span>
                +32%
            </span>
        </div>
    </div>
</section>,
  "nodes": Array [
    <section
      className="aggregator"
>
      <div
            className="aggregator__container"
      >
            <h1
                  className="aggregator__pl-currency"
            >
                  <Indicator
                        className="aggregator__pl-indicator"
                        inProfit={true}
                  />
                  <span>
                        £193.94
                  </span>
            </h1>
            <div
                  className="aggregator__metadata"
            >
                  <span>
                        £606.06 paid
                  </span>
                  <span>
                        £800 now
                  </span>
                  <span>
                        +32%
                  </span>
            </div>
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Aggregator
        activeWallet={
                Object {
                        "balance": Object {
                          "amount": "800",
                          "currency": "GBP",
                        },
                        "currency": "BTC",
                        "id": "58542935-67b5-56e1-a3f9-42686e07fa40",
                        "name": "BTC",
                        "purchases": Array [
                          Object {
                            "currency": "BTC",
                            "id": "9e14d574-30fa-5d85-b02c-6be0d851d61d",
                            "total": Object {
                              "amount": "101.01",
                              "currency": "GBP",
                            },
                          },
                          Object {
                            "currency": "BTC",
                            "id": "8e14d574-30fa-5d85-b02c-6be0d851d61d",
                            "total": Object {
                              "amount": "202.02",
                              "currency": "GBP",
                            },
                          },
                          Object {
                            "currency": "BTC",
                            "id": "8e14d574-30fa-5d85-b02c-6be0d851d61d",
                            "total": Object {
                              "amount": "303.03",
                              "currency": "GBP",
                            },
                          },
                        ],
                      }
        }
        isLoading={false}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Aggregator {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "defaults": Object {
          "balanceNow": "",
          "inProfit": true,
          "percentage": "",
          "profitLoss": "",
          "totalPaid": "",
        },
        "props": Object {
          "activeWallet": Object {
            "balance": Object {
              "amount": "800",
              "currency": "GBP",
            },
            "currency": "BTC",
            "id": "58542935-67b5-56e1-a3f9-42686e07fa40",
            "name": "BTC",
            "purchases": Array [
              Object {
                "currency": "BTC",
                "id": "9e14d574-30fa-5d85-b02c-6be0d851d61d",
                "total": Object {
                  "amount": "101.01",
                  "currency": "GBP",
                },
              },
              Object {
                "currency": "BTC",
                "id": "8e14d574-30fa-5d85-b02c-6be0d851d61d",
                "total": Object {
                  "amount": "202.02",
                  "currency": "GBP",
                },
              },
              Object {
                "currency": "BTC",
                "id": "8e14d574-30fa-5d85-b02c-6be0d851d61d",
                "total": Object {
                  "amount": "303.03",
                  "currency": "GBP",
                },
              },
            ],
          },
          "isLoading": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="aggregator"
>
          <div
                    className="aggregator__container"
          >
                    <h1
                              className="aggregator__pl-currency"
                    >
                              <Indicator
                                        className="aggregator__pl-indicator"
                                        inProfit={true}
                              />
                              <span>
                                        £193.94
                              </span>
                    </h1>
                    <div
                              className="aggregator__metadata"
                    >
                              <span>
                                        £606.06 paid
                              </span>
                              <span>
                                        £800 now
                              </span>
                              <span>
                                        +32%
                              </span>
                    </div>
          </div>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="aggregator"
>
          <div
                    className="aggregator__container"
          >
                    <h1
                              className="aggregator__pl-currency"
                    >
                              <Indicator
                                        className="aggregator__pl-indicator"
                                        inProfit={true}
                              />
                              <span>
                                        £193.94
                              </span>
                    </h1>
                    <div
                              className="aggregator__metadata"
                    >
                              <span>
                                        £606.06 paid
                              </span>
                              <span>
                                        £800 now
                              </span>
                              <span>
                                        +32%
                              </span>
                    </div>
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Aggregator
    activeWallet={
        Object {
            "balance": Object {
              "amount": "800",
              "currency": "GBP",
            },
            "currency": "BTC",
            "id": "58542935-67b5-56e1-a3f9-42686e07fa40",
            "name": "BTC",
            "purchases": Array [
              Object {
                "currency": "BTC",
                "id": "9e14d574-30fa-5d85-b02c-6be0d851d61d",
                "total": Object {
                  "amount": "101.01",
                  "currency": "GBP",
                },
              },
              Object {
                "currency": "BTC",
                "id": "8e14d574-30fa-5d85-b02c-6be0d851d61d",
                "total": Object {
                  "amount": "202.02",
                  "currency": "GBP",
                },
              },
              Object {
                "currency": "BTC",
                "id": "8e14d574-30fa-5d85-b02c-6be0d851d61d",
                "total": Object {
                  "amount": "303.03",
                  "currency": "GBP",
                },
              },
            ],
          }
    }
    isLoading={false}
/>,
}
`;
